core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -871,6 +871,55 @@ public void deleteTenant(int tenantId) throws UserStoreException {
         }
     }
 
+    public void deleteTenant(String tenantUniqueID) throws UserStoreException {
+
+        try {
+            deleteTenant(tenantUniqueID, true);
+        } catch (UserStoreException e) {
+            throw new UserStoreException(e);
+        }
+    }
+
+    /**
+     * Delete Tenant
+     *
+     * @param tenantUniqueID              - Tenant Id
+     * @param removeFromPersistentStorage - Flag to decide weather delete from persistent storage
+     * @throws UserStoreException
+     */
+    public void deleteTenant(String tenantUniqueID, boolean removeFromPersistentStorage)
```

Comment:
What is the purpose of supporting non-persistent tenant deletion? 

Will that tenant available after a restart?

Reply:
Yes only the cache will be deleted.